ta/pkcs11/src/attributes.h

```diff
@@ -262,6 +262,25 @@ static inline enum pkcs11_key_type get_key_type(struct obj_attrs *head)
 	return type;
 }
 
+/*
+ * get_certificate_type() - Get the certificate type of an object
+ * @head:	Pointer to serialized attributes
+ *
+ * Returns the certificate type of an object on success or returns
+ * PKCS11_CKC_UNDEFINED_ID on error.
+ */
+static inline
+enum pkcs11_certificate_type get_certificate_type(struct obj_attrs *head)
+{
+	uint32_t type = 0;
+	uint32_t size = sizeof(type);
+
+	if (get_attribute(head, PKCS11_CKA_CERTIFICATE_TYPE, &type, &size))
+		return PKCS11_CKC_UNDEFINED_ID;
+
```

Comment:
minor: `assert(size == sizeof(uint32_t));`
or I think you could even drop fetching size.

Reply:
I feel better if we fetch it ;) ... but will add the assert

Reply:
oh -- none of the other methods with the same construct have it -- should we add assert to those too?

Reply:
`get_u32_attribute()` would have sanity check embedded.

Reply:
I agree, `get_u32_attribute()` and no extra assertions.

Reply:
Updated the PR with that.